http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -49,17 +58,67 @@
       onMethod_ = {@VisibleForTesting})
   private boolean connectionIsOpen = false;
 
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  private HttpClient httpClient = HttpClient.newHttpClient();
+
   private final URI serverUri;
 
   private boolean closed = false;
 
+  private WebSocket client;
+
   @Getter(
       value = AccessLevel.PACKAGE,
       onMethod_ = {@VisibleForTesting})
-  @Setter(
-      value = AccessLevel.PACKAGE,
-      onMethod_ = {@VisibleForTesting})
-  private WebSocketClient client;
+  private final WebSocket.Listener webSocketListener =
+      new Listener() {
+        private final StringBuilder textAccumulator = new StringBuilder();
+
+        @Override
+        public void onOpen(final WebSocket webSocket) {
+          synchronized (queuedMessages) {
+            client = webSocket;
+            connectionIsOpen = true;
+            queuedMessages.forEach(
+                message ->
+                    client
+                        .sendText(message, true)
+                        .exceptionally(logWebSocketError("Failed to send queued text.")));
+            queuedMessages.clear();
+          }
+          webSocket.request(1);
+        }
+
+        @Override
+        public CompletionStage<?> onText(
+            final WebSocket webSocket, final CharSequence data, final boolean last) {
```

Comment:
I like that we continue to abstract away the underlying API, though `CharSequence` and `boolean last` seem to be not as nice of an API. I think this websocket library loses points here.

Reply:
Not a real fan of it either, but by looking at the documentation it seems that the API in theory allows you to do everything async, including message processing of messages that get split.

I actually believe we could ignore the last boolean. When debugging some issue I was having (due to wrong usage of the API) I realized that the data messages seem complete for the most part.